spyder/widgets/panels/line_number.py

```diff
@@ -0,0 +1,215 @@
+                        draw_pixmap(top, self.bp_pixmap)
+                    else:
+                        draw_pixmap(top, self.bpc_pixmap)
+
+    def mouseMoveEvent(self, event):
+        """Override Qt method.
+
+        Show code analisis, if left button pressed select lines.
+        """
+        line_number = self.editor.get_linenumber_from_mouse_event(event)
+        block = self.editor.document().findBlockByNumber(line_number-1)
+        data = block.userData()
+
+        # this disables pyflakes messages if there is an active drag/selection
+        # operation
+        check = self._released == -1
+        if data and data.code_analysis and check:
+            self.editor.show_code_analysis_results(line_number, data.code_analysis)
+
+        if event.buttons() == Qt.LeftButton:
+            self._released = line_number
+            self.editor.select_lines(self._pressed, self._released)
+
+
+    def mouseDoubleClickEvent(self, event):
+        """Override Qt method.
+
+        Add or remove breakpoints.
+        """
+        line_number = self.editor.get_linenumber_from_mouse_event(event)
+        shift = event.modifiers() & Qt.ShiftModifier
+        self.editor.add_remove_breakpoint(line_number, edit_condition=shift)
+
+
+    def mousePressEvent(self, event):
+        """Override Qt method
+
+        Select line, and starts selection
+        """
+        line_number = self.editor.get_linenumber_from_mouse_event(event)
+        self._pressed = line_number
+        self._released = line_number
+        self.editor.select_lines(self._pressed,
+                                         self._released)
+
+    def mouseReleaseEvent(self, event):
+        """Override Qt method."""
+        self._released = -1
+        self._pressed = -1
+
+    def wheelEvent(self, event):
+        """Override Qt method."""
+        self.editor.wheelEvent(event)
+
+    # --- Other methods
+    # -----------------------------------------------------------------
+
+    def compute_width(self):
+        """Compute and return line number area width"""
+        if not self._enabled:
+            return 0
+        digits = 1
+        maxb = max(1, self.editor.blockCount())
+        while maxb >= 10:
+            maxb /= 10
+            digits += 1
+        if self._margin:
+            margin = 3+self.editor.fontMetrics().width('9'*digits)
+        else:
+            margin = 0
+        return margin+self.get_markers_margin()
+
+    def get_markers_margin(self):
+        if self._markers_margin:
+            return self._markers_margin_width
+        else:
+            return 0
+
+    def update_width(self, new_block_count=None):
+        """
+        Update line number area width.
+
+        new_block_count is needed to handle blockCountChanged(int) signal
+        """
+        self.editor.setViewportMargins(self.compute_width(), 0,
+                                self.editor.get_scrollflagarea_width(), 0)
+
+    def update_linenumberarea(self, qrect, dy):
```

Comment:
There is a function called update in qwidget class http://doc.qt.io/qt-5/qwidget.html#update, because of that I didn't change this name, should I override this function?